In [1]:
from tools_configs import *

# Read frequency word list

In [2]:

wordset_freq = {}
wordset = set()

with open("wordlists/chinese_charfreq_simpl_trad.txt", "r", encoding="utf-8") as fread:
    next(fread)
    contents = fread.read()

    wordset.update(contents)

    if "\n" in wordset:
        wordset.remove("\n")

    if " " in wordset:
        wordset.remove(" ")

    wordset_freq = {word: num + 1 for num, word in enumerate(contents.split("\n"))}

def find_freq(word):
    return wordset_freq[word] if word in wordset_freq else BIGNUM


def sort_by_freq(list_chars, reverse=False):
    if not list_chars:
        return []

    items = sorted(
        [(word, find_freq(word)) for word in list_chars], key=lambda x: (x[1], x[0]), reverse=reverse
    )

    return [word for word, order in items]

rads = Radicals()

rads.load_radical_data()

import json
try:
    with open('char_dict.json', "r", encoding="utf-8") as fread:
        char_dict = json.load(fread)
except:
    print(f'No file {'char_dict.json'}')

print(f'{len(char_dict)=}')
wordset = set()

with open("wordlists/chinese_charfreq_simpl_trad.txt", "r", encoding="utf-8") as fread:
    next(fread)
    contents = fread.read()

    wordset.update(contents)

    if "\n" in wordset:
        wordset.remove("\n")

    if " " in wordset:
        wordset.remove(" ")

    wordset_freq = {word: num + 1 for num, word in enumerate(contents.split("\n"))}

print(f'{len(wordset_freq)=}')
exiting_word = [word for word in wordset if word in char_dict]
print(f'{len(exiting_word)=}')

len(char_dict)=13056
len(wordset_freq)=12042
len(exiting_word)=11995


# Count
- No. Words by radical count
    
    - Most popular radicals

- Number and percents of words by radical count

- Check words agains top freqent words (top 12k)




In [3]:
chars_by_rad_count = {}
chars_by_radical = {}
most_freq_radicals = {}

for word in exiting_word:
    item = char_dict[word]

    components = item['components']

    count = 0
    for comp in components:
        if rads.is_radical_variant(comp):
            count += 1

            chars_by_radical.setdefault(rads.norminal(comp), [])
            chars_by_radical[rads.norminal(comp)].append(word)

    chars_by_rad_count.setdefault(count, [])
    chars_by_rad_count[count].append(word)

count_chars_by_radical = sorted([(key, len(value)) for key, value in chars_by_radical.items()], key = lambda x: x[1], reverse=True)
count_chars_by_rad_count = sorted([(key, len(value)) for key, value in chars_by_rad_count.items()], key = lambda x: x[1], reverse=True)

print('Most freq radical count')
for num, item in enumerate(count_chars_by_rad_count):
    print(f'{item[0]}\t{item[1]}')

print('Most freq radicals')
for num, item in enumerate(count_chars_by_radical):
    print(f'{item[0]}\t{item[1]}')
    most_freq_radicals[item] = num+1

freq_chars_by_rad_count_stats = {}
practical_chars = set()
char_components = {}
radicals = rads.radical_list

char_set_by_radical = {}
for current in radicals:
    char_set_by_radical.setdefault(current, set())

for i in range(1, 6):
    words = chars_by_rad_count[i]
    freq_chars_by_rad_count_stats[i] = words
    practical_chars.update(words)

    for word in words:
        char_components.setdefault(word, [])

        components = char_dict[word]['components']

        for comp in components:
            if not rads.is_radical_variant(comp):
                continue

            norminal = rads.norminal(comp)
            char_set_by_radical[norminal].add(word)

            char_components[word].append(norminal)

        char_components[word].sort()
        pass
print(freq_chars_by_rad_count_stats.keys())

print(f'{count=}')
print(f'{len(practical_chars)=}')
print(f'{len(char_components)=}')
print(f'{len(char_set_by_radical)=}')


Most freq radical count
3	4290
4	2690
2	2605
5	1130
6	569
1	425
7	150
0	101
8	33
11	1
10	1
Most freq radicals
⼀	1931
⼝	1780
⼋	1300
⼁	1292
⾋	1189
⼈	1086
⽇	929
⼗	820
⼆	753
⽊	725
⼂	723
⽉	705
⾦	672
⽥	594
⽔	551
⽷	533
⼃	519
⾔	475
⼑	465
⼟	452
⼇	448
⼤	425
⾍	421
⼩	420
⼼	413
⼌	399
⼛	392
⼿	369
⼜	365
⾩	342
⼧	329
⽵	319
⿃	315
⼓	314
⽬	314
⿂	312
⼍	311
⼄	306
⾙	297
⼥	288
⽕	283
⾡	277
⽯	277
⼡	264
⼚	257
⾐	249
⽲	248
⼉	248
⼎	246
⼕	237
⾞	233
⾫	227
⾜	224
⾺	219
⽟	215
⽰	213
⼽	212
⼘	210
⼙	210
⽁	201
⼫	199
⼭	195
⽶	195
⼹	185
⾨	183
⽌	180
⽹	177
⼞	173
⼔	173
⼅	168
⼨	166
⼒	165
⼲	164
⾴	160
⽻	153
⼱	153
⼠	153
⾷	151
⼦	147
⽿	145
⼐	143
⾬	141
⽴	136
⽺	133
⽝	133
⽩	128
⼯	126
⼶	125
⽖	124
⾣	121
⾅	116
⽫	113
⼴	113
⾌	109
⽳	108
⼏	106
⼳	101
⼰	100
⾰	100
⽧	100
⽤	99
⽋	99
⼣	99
⼺	98
⼸	97
⾑	96
⾀	93
⽜	93
⽄	89
⽢	86
⾈	85
⽅	82
⽎	80
⽮	78
⽼	77
⾖	72
⽭	68
⼪	68
⼻	67
⼾	67
⾟	66
⼬	66
⾤	65
⾽	64
⾒	64
⾗	62
⾥	61
⽍	60
⽐	60
⽽	58
⾮	56
⼮	56
⽚	56
⼖	55
⽂	53
⾭	51
⿅	49
⾛	49
⾓	49
⽸	48
⾕	47
⽈	47
⽒	47
⽱	45
⾉	44
⽏	43
⼵	43
⾻	42
⾘	42
⾳	40
⿒	40
⿊	39
⾏	38
⾱	38
⿓	37
⿁	35
⾚	35

# Build the tree

In [4]:
from treelib import Node, Tree
import itertools


tree = Tree()

base = tree.create_node('', '')

for current in radicals:
    if current in char_set_by_radical:
        char_set = char_set_by_radical[current]

        if char_set:
            node = tree.create_node(identifier=current, tag=current, parent=base, data = char_set)

# tree.show()
# for pre, _, node in RenderTree(root):
#     print("%s%s %i" % (pre, node.name, len(node.chars)))


def add_item(components, current):
    if len(components) < 1:
        return
    
    comp = components[0]

    # print(parent)

    child_comps = current.data - char_set_by_radical[comp]

    if child_comps:
        
        child_id = current.identifier+comp
        
        node=tree.get_node(child_id)

        if not (node):
            node = tree.create_node(identifier=child_id, tag=child_id, parent=current, data=child_comps)
        
        add_item(components[1:], node)

    # tree.show()

count = 0

for word in char_components:
    components = char_components[word]
    if len(char_components[word]) < 2:
        continue
    
    current = tree.get_node(components[0])
    add_item(''.join(components[1:]), current)
    count += 1

    # if count > 500:
    #     break
    pass

tree.show()
tree_string = tree.show(stdout=False)
print(f'{len(tree.nodes)=}')


├── ⼀
│   ├── ⼀⼁
│   │   ├── ⼀⼁⼂
│   │   │   └── ⼀⼁⼂⼃
│   │   │       ├── ⼀⼁⼂⼃⼝
│   │   │       ├── ⼀⼁⼂⼃⽫
│   │   │       ├── ⼀⼁⼂⼃⾒
│   │   │       └── ⼀⼁⼂⼃⾦
│   │   ├── ⼀⼁⼃
│   │   │   ├── ⼀⼁⼃⼄
│   │   │   │   ├── ⼀⼁⼃⼄⼿
│   │   │   │   ├── ⼀⼁⼃⼄⽅
│   │   │   │   ├── ⼀⼁⼃⼄⽶
│   │   │   │   ├── ⼀⼁⼃⼄⽷
│   │   │   │   ├── ⼀⼁⼃⼄⾡
│   │   │   │   └── ⼀⼁⼃⼄⾩
│   │   │   ├── ⼀⼁⼃⼆
│   │   │   │   ├── ⼀⼁⼃⼆⼈
│   │   │   │   ├── ⼀⼁⼃⼆⼝
│   │   │   │   ├── ⼀⼁⼃⼆⼼
│   │   │   │   ├── ⼀⼁⼃⼆⽇
│   │   │   │   ├── ⼀⼁⼃⼆⽉
│   │   │   │   ├── ⼀⼁⼃⼆⽊
│   │   │   │   ├── ⼀⼁⼃⼆⽕
│   │   │   │   ├── ⼀⼁⼃⼆⽧
│   │   │   │   ├── ⼀⼁⼃⼆⽯
│   │   │   │   ├── ⼀⼁⼃⼆⽰
│   │   │   │   ├── ⼀⼁⼃⼆⽳
│   │   │   │   ├── ⼀⼁⼃⼆⽵
│   │   │   │   ├── ⼀⼁⼃⼆⾈
│   │   │   │   ├── ⼀⼁⼃⼆⾋
│   │   │   │   ├── ⼀⼁⼃⼆⾍
│   │   │   │   ├── ⼀⼁⼃⼆⾔
│   │   │   │   ├── ⼀⼁⼃⼆⾡
│   │   │   │   ├── ⼀⼁⼃⼆⾣
│   │   │   │   ├── ⼀⼁⼃⼆⾩
│   │   │   │   ├── ⼀⼁⼃⼆⿂
│   │   │   │   └── ⼀⼁⼃⼆⿒
│   │   │   └── ⼀⼁⼃⼗
│   │   ├── ⼀⼁⼆
│   │   │   ├── ⼀⼁⼆⼈
│   │   │   │   ├── ⼀⼁⼆

In [5]:
components

['⼍', '⼗', '⼜', '⽖', '⽟']

# Alternative chars for box drawing

'├' => '＼'
'└' => '＼'
'─' => '⼀'
' ' => '　' 

```
Current
＼⼀⼀　⼀
｜　　　＼⼀⼀　⼀⼁
｜　　　｜　　　＼⼀⼀　⼀⼁⼂
｜　　　｜　　　｜　　　＼⼀⼀　⼀⼁⼂⼃
｜　　　｜　　　｜　　　　　　　＼⼀⼀　⼀⼁⼂⼃⼝
｜　　　｜　　　｜　　　　　　　＼⼀⼀　⼀⼁⼂⼃⽫
｜　　　｜　　　｜　　　　　　　＼⼀⼀　⼀⼁⼂⼃⾒
｜　　　｜　　　｜　　　　　　　＼⼀⼀　⼀⼁⼂⼃⾦
｜　　　｜　　　＼⼀⼀　⼀⼁⼃
｜　　　｜　　　｜　　　＼⼀⼀　⼀⼁⼃⼄
｜　　　｜　　　｜　　　｜　　　＼⼀⼀　⼀⼁⼃⼄⼿
｜　　　｜　　　｜　　　｜　　　＼⼀⼀　⼀⼁⼃⼄⽅

# Originals
├── ⼀
│   ├── ⼀⼁
│   │   ├── ⼀⼁⼂
│   │   │   └── ⼀⼁⼂⼃
│   │   │       ├── ⼀⼁⼂⼃⼝
│   │   │       ├── ⼀⼁⼂⼃⽫
│   │   │       ├── ⼀⼁⼂⼃⾒
│   │   │       └── ⼀⼁⼂⼃⾦
│   │   ├── ⼀⼁⼃
│   │   │   ├── ⼀⼁⼃⼄
│   │   │   │   ├── ⼀⼁⼃⼄⼿
│   │   │   │   ├── ⼀⼁⼃⼄⽅
```

# Reformat box drawing to fullwidth

In [6]:
def get_chars_from_rads(radicals):
    if not radicals:
        return None

    keys = list(radicals)
    lenkey = len(keys)


    char_set = char_set_by_radical[keys[0]]
    
    # print(f'{len(char_set)=}')

    for key in keys[1:]:
        char_set = char_set & char_set_by_radical[key]

        # print(f'{len(char_set)=}')

    return char_set

# Replacement for fullwidth box drawing characters 
def hex_show(char):
    return '\\u' + hex(ord(char))[2:].upper()

def replace_box_chars(tree):
    new_tree = tree
    char_map = {
        '├' : '＼',
        '└' : '＼',
        '─' : '⼀',
        ' ' : '　',
    }

    for char in char_map:
        new_tree = new_tree.replace(char, char_map[char])

    return new_tree

tree_str = '''
├── ⼀
│   ├── ⼀⼁
│   │   ├── ⼀⼁⼂
│   │   │   └── ⼀⼁⼂⼃
│   │   │       ├── ⼀⼁⼂⼃⼝
│   │   │       ├── ⼀⼁⼂⼃⽫
│   │   │       ├── ⼀⼁⼂⼃⾒'''


# print(f'[├] => [＼] {hex_show('├')} => {hex_show('＼')}')
# print(f'[└] => [＼] {hex_show('└')} => {hex_show('＼')}')
# print(f'[─] => [⼀] {hex_show('─')} => {hex_show('⼀')}')
# print(f'[ ] => [　] {hex_show(' ')} => {hex_show('　')}')

# print(f'tree: \n{tree_str}\nnew tree: \n{replace_box_chars(tree_str)}')



# Build the dictionary

In [8]:
lines = tree_string.split('\n')
key_to_sorted_chars = {}

def replace_key(match_obj):
    key = match_obj.group(0)

    if key not in key_to_sorted_chars:
        chars = sort_by_freq(get_chars_from_rads(key))
        key_to_sorted_chars[key] = chars
    else:
        chars = key_to_sorted_chars[key]

    return f'{key[-1]}: {''.join(chars[:10])} {len(chars)}' 

def replace_simple_key(match_obj):
    key = match_obj.group(0)

    if key not in key_to_sorted_chars:
        chars = sort_by_freq(get_chars_from_rads(key))
        key_to_sorted_chars[key] = chars
    else:
        chars = key_to_sorted_chars[key]

    return f'{key[-1]}: {''.join(chars[:10])} {len(chars)}' 

# Build map between keys and chars at a node
for key in tree.nodes:
   
    if key not in key_to_sorted_chars:
        chars = sort_by_freq(get_chars_from_rads(key))
        key_to_sorted_chars[key] = chars
    else:
        chars = key_to_sorted_chars[key]
    pass
pass

HOME_PAGE = "偏旁"
HOME_PAGE_PY = "piānpáng"

string = ''
HOME_PAGE = "偏旁"
HOME_PAGE_PY = "piānpáng"

back_home = f"""{pleco_make_dark_gray('HOMEPAGE')} {PC_RIGHT_TRIANGLE} {pleco_make_link(HOME_PAGE)}\n"""

fpleco = open('radical_lookup_pleco.txt', 'w', encoding='utf-8')
string+=back_home

def get_rad_pinyin(rad_list):
    return ' '.join([rads.lookup(rad)['pinyin'] for rad in rad_list])

def get_rad_pinyin_abbrev(rad_list):
    return ' '.join([rads.lookup(rad)['pinyin'][0] for rad in rad_list])

MAX_SHOW = 5
WIDE_SPACE = '\u3000'

count = 0

def node_to_pleco(current):
    children = sorted([x.identifier for x in tree.children(current)])
    current_chars = key_to_sorted_chars[current]
    len_current = len(current)
    
    global count

    count += 1

    if count > 30:
        return
    
    string = ''
    
    if not current: # This is root
        key_perms = [HOME_PAGE]
        key_py_perms = [HOME_PAGE_PY]
    else:
        key_perms = [''.join(item) for item in itertools.permutations(current)]
        key_py_perms = [get_rad_pinyin(key) for key in key_perms]

    # string = f'\t{key_py}\t'

    if current_chars:
        only_these_radicals = [char for char in current_chars if len(char_components[char]) == len_current]
        count_str = ''
        to_show = len(only_these_radicals) if not children else MAX_SHOW # If no more child branches, show everything

        if to_show > MAX_SHOW:
            count_str = f'+{to_show-MAX_SHOW}'

        shown_chars = [pleco_make_link(char) for char in only_these_radicals[:to_show]]
        num_chars = pleco_make_dark_gray(f'FOUND CHARACTERS: {len(only_these_radicals)}')

        if len(current_chars) == 1:
            string += f'{num_chars}\n{f'{PC_MIDDLE_DOT}'.join(shown_chars)}\n'
        else:
            string += f'{num_chars}\n{f'{PC_MIDDLE_DOT}'.join(shown_chars)} {count_str}\n'
        tree

    if children:
        string += pleco_make_dark_gray(f'BRANCHES: {len(children)}\n')

        for child in children:
            if not child:
                continue

            child_chars = key_to_sorted_chars[child]
            child_children = sorted([x.identifier for x in tree.children(child)])

            child_to_show = len(child_chars) if not child_children else MAX_SHOW # If no more child branches, show everything

            shown_chars = [pleco_make_link(char) for char in child_chars[:child_to_show]]
            
            if len(child_chars) == 1:
                actual_components = ''.join(char_components[child_chars[0]])
                string += f'{pleco_make_link(actual_components)}{WIDE_SPACE}{f'{PC_MIDDLE_DOT}'.join(shown_chars)}\n'
            else:
                string += f'{pleco_make_link(child)}{WIDE_SPACE}{f'{PC_MIDDLE_DOT}'.join(shown_chars)} ({len(child_children)}/{len(child_chars)})\n'

    ''' Currently does not work 
    if len(current) > 1:
        step_back = f"""{pleco_make_dark_gray('STEP BACK')} {PC_LEFT_TRIANGLE} {pleco_make_link(current[1:])}\n"""
        string += step_back
    '''
    
    if current:
        string += back_home
        
    string = string.replace('\n', PC_NEW_LINE)
    
    for key, py in zip(key_perms, key_py_perms):
        fpleco.write(f'{key}\t{py}\t{string}\n')

    for child in children:
        if not child:
            continue

        node_to_pleco(child)

node_to_pleco('')

string = string.replace('\n', PC_NEW_LINE)
fpleco.write(f'{string}\n')

fpleco.close()

In [9]:
# nodes = tree.nodes

count = 0
total = set()
list_py = []
set_py = set()
abb_list_py = []
abb_set_py = set()

for key in tree.nodes:
    if not key:
        continue

    if len(key) < 2:
        continue
    #  if len(tree.get_node(key).data) != 1:
    #     continue

    py = get_rad_pinyin(key)
    list_py.append(py)
    set_py.add(py)

    abb_py = get_rad_pinyin_abbrev(key)
    abb_list_py.append(abb_py)
    abb_set_py.add(abb_py)

    # if py == 'yī jīn':
    #   print(key)
    #   print(tree.get_node(key).data)

print(f"{len(list_py)}")
print(f"{len(abb_set_py)}")

print(f"{len(abb_list_py)}")
print(f"{len(abb_set_py)}")
# print(f'{len(total)}')


def list_duplicates(seq):
    seen = set()
    seen_add = seen.add
    # adds all elements it doesn't know yet to seen and all other to seen_twice
    seen_twice = set(x for x in seq if x in seen or seen_add(x))
    # turn the set into a list (as requested)
    return list(seen_twice)


dupl = list_duplicates(list_py)
print(dupl)  # yields [1, 2, 5]
print(f"{len(dupl)}")

abb_dupl = list_duplicates(abb_list_py)
print(abb_dupl)  # yields [1, 2, 5]
print(f"{len(abb_dupl)}")


8623
4714
8623
4714
['tǔ mù', 'yī jīn', 'kǒu zhǐ shí', 'kǒu gōng', 'jīn rì', 'jīn zhuī', 'bǔ jīn', 'yòu shí', 'shí shān yuè', 'bā dāo hǎn mù', 'shí zhuī', 'sī gōng mì', 'yī piě mù', 'bā dāo hǎn mù yán', 'fāng yán', 'mù lóng', 'shān shì', 'bǐ fù', 'shì yī', 'rén lì', 'máo biāo', 'rì shì', 'xiǎo wāng jīn', 'gān yǒu', 'shǒu mù', 'yī shí mù', 'yòu mù', 'gōng shuǐ', 'jīn lì', 'piě xiǎo mù', 'rén shì bèi', 'shí zú', 'mù zhuī', 'bīng shí', 'huǒ shí', 'zhǔ rén sī yǔ', 'chuān mù', 'rén bǐ', 'gǔ niǎo', 'bǐ hé', 'rén shān', 'yī rén xiǎo mù', 'sī gōng', 'chì yù', 'bǔ kǒu jīn', 'jīn mù', 'kǒu yuè mù', 'pū mù', 'yī gǔn èr cùn jīn', 'mù shì', 'gān fù', 'bǐ shí', 'tóu bā shí', 'mù zhì', 'rén wéi', 'shuǐ mǐn', 'mù zhú', 'kǒu yǎn yù', 'gān chóng', 'shí cǎo', 'xiǎo mù', 'shǒu wú', 'yī jiōng kǒu shí', 'mì kǒu tǔ xiǎo mù', 'yī zhǐ', 'lì niǎo', 'shí zhǐ', 'sī gōng shuǐ', 'rén shì', 'shuǐ yù', 'ér bīng mù', 'yī mì', 'shí kǒu shān', 'mù yáng', 'gǔn bā gōng mù', 'mù yǔ', 'qiàn shí', 'nǚ lì yǔ', 'yī piě jīn', '

In [10]:
print('\n'.join(abb_list_py))

s c
w c
r f
c g
c q
n g
s y
s s
ě y
c g
k q
n j
s b
m j
z ě
c f
r j
s b
h y
h l
s y
z z
s y
q q
r m
j q
y b
q z
p f
k m
m c
m c
m m
z z
c l
y p
s b
z m
j g
y q
p z
j b
b j
w s
m s
b c
x n
s y
b r
y m
s m
k m
s m
f s
z m
s z
m c
s g
p y
m y
g x
n w
m m
r c
b k
s c
c f
j y
s s
f m
d j
w y
m s
h y
l z
j y
g f
g c
y z
l l
f m
w m
b y
q s
f f
l n
q h
x m
j s
y m
r y
b x
t h
g z
h y
ě m
s x
z g
m t
g f
m m
p y
b j
y j
w y
f s
j y
s y
s s
x s
r s
b s
l k
s c
p z
m s
y m
k l
s s
r b
r ě
g c
s q
r s
s g
g g
w r
s l
m m
x m
y f
c y
b n
s z
y j
x z
y s
c t
y ě
c s
p m
x j
m l
b z
x c
m m
y c
è d
x y
s k
z c
r y
x x
p j
m y
f k
è s
y g
g s
b g
n h
k m
d n
n y
k c
y b
c m
z m
d x
s x
c y
g y
g z
s l
k x
n f
p s
b y
m y
d z
s m
y g
c f
z j
w c
z j
y g
q g
k s
b l
g h
y y
j h
s y
y ě
b c
b n
r q
p r
j s
s s
c m
q s
d c
j c
s c
t z
g x
h q
t m
m c
b k
m b
s m
y s
s z
r n
y n
é y
b m
c q
r s
s y
b y
p c
r l
s m
m j
y l
p c
d q
j y
y l
y z
m f
d y
é n
s x
s y
k l
w m
t j
g d
z j
c l
g n
s m
h f
j g
y j


In [11]:
print('\n'.join(list_py))

shǐ chuò
wéi chì
rén fēi
cǎo gǔ
chóng qí
niú gāo
shuǐ yáng
shí shǔ
ěr yán
cǎo gāo
kǒu quǎn
nǚ jǐ
shuǐ bí
mǐ jīn
zhǐ ěr
cǎo fēi
rì jīn
shǒu bèi
hū yán
hǎn lì
shǒu yì
zhì zú
shuǐ yǒu
qì qīng
rì mén
jiàn qīng
yǎn bǐ
quǎn zì
pū fāng
kǒu mǎ
máo cǎo
mù cǎo
mì má
zhǔ zhǐ
chē lù
yuè pǐ
shǒu bái
zhǔ mù
jué gān
yǔ qí
piě zú
jīn bèi
bèi jīn
wú shí
mián shí
bái chuò
xué niǎo
shū yáng
bǐ rì
yán mén
shī máo
kǒu mián
shān mù
fāng shǐ
zhī mù
shǐ zhuī
mǐn chóng
shí gāo
piě yán
mù yù
gōng xué
nǚ wāng
mù máo
rén cháng
bā kǒu
shū cǎo
chì fāng
jī yòu
sī shǐ
fēi má
dà jīn
wǎng yán
mù shuǐ
huǒ yáng
lì zhú
jǐ yán
gǔ fù
gèn cǎo
yán zhuī
lì lǐ
fāng mì
wén mì
bǐ yuè
qì shuǐ
fāng fēi
lì niǎo
quǎn hēi
xuán mù
jì shuǐ
yī mén
rì yuè
bāo xīn
tǔ hēi
gān zhú
hé yú
ěr mén
shuǐ xíng
zhú gāo
mù tián
gōng fù
mì mǐn
piàn yú
bīng jīn
yù jiàn
wén yù
fāng shǒu
jiù yī
shì yuè
shí shǒu
xīn shé
rù shuǐ
bǐ shí
lì kǒu
sī cǎo
pí zú
mián shì
yuè mén
kǒu lóng
shé shí
rén bǐ
rén ěr
gèn chuò
shí qīng
rén shì
shuǐ gǔ
gǔ gāo
wāng rì
shì l

# Distributions of radicals
| Len | Count | Pecentitle |
|-----|-------|------------|
| 0   | 149   | 1.2%       |
| 1   | 724   | 7.3%       |
| 2   | 2728  | 30.0%      |
| 3   | 4268  | 65.6%      |
| 4   | 2608  | 87.3%      |
| 5   | 1004  | 95.7%      |
| 6   | 357   | 98.7%      |
| 7   | 73    | 99.3%      |
| 8   | 59    | 99.8%      |
| 9   | 22    | 100.0%     |
| 10  | 3     | 100.0%     |
| Sum | 11995 |


So we will focus on words with 5 or less radicals, remove words without radicals

# Most frequent rads

| Pos | Radical | Char count | Percentitle |
|-----|---------|------------|-------------|
| 1   | ⼀       | 2015       | 5.3%        |
| 2   | ⼝       | 1668       | 9.7%        |
| 3   | ⾋       | 1222       | 13.0%       |
| 4   | ⼁       | 1151       | 16.0%       |
| 5   | ⼈       | 1025       | 18.7%       |
| 6   | ⽇       | 918        | 21.2%       |
| 7   | ⼆       | 795        | 23.3%       |
| 8   | ⼂       | 786        | 25.4%       |
| 9   | ⼋       | 722        | 27.3%       |
| 10  | ⽉       | 689        | 29.1%       |
| 11  | ⼗       | 674        | 30.9%       |
| 12  | ⽊       | 669        | 32.6%       |
| 13  | ⾦       | 666        | 34.4%       |
| 14  | ⽔       | 545        | 35.8%       |
| 15  | ⽷       | 532        | 37.3%       |
| 16  | ⼃       | 519        | 38.6%       |
| 17  | ⾔       | 475        | 39.9%       |
| 18  | ⼑       | 463        | 41.1%       |
| 19  | ⼄       | 450        | 42.3%       |
| 20  | ⼇       | 448        | 43.5%       |
| 21  | ⼤       | 425        | 44.6%       |
| 22  | ⼩       | 420        | 45.7%       |
| 23  | ⾍       | 408        | 46.8%       |
| 24  | ⼼       | 408        | 47.9%       |
| 25  | ⼌       | 406        | 49.0%       |
| 26  | ⼟       | 396        | 50.0%       |
| 27  | ⼿       | 357        | 50.9%       |
| 28  | ⾩       | 342        | 51.9%       |
| 29  | ⼜       | 341        | 52.8%       |
| 30  | ⽥       | 337        | 53.6%       |
| 31  | ⼧       | 329        | 54.5%       |
| 32  | ⽵       | 319        | 55.4%       |
| 33  | ⿃       | 315        | 56.2%       |
| 34  | ⼓       | 314        | 57.0%       |
| 35  | ⿂       | 312        | 57.9%       |
| 36  | ⼍       | 311        | 58.7%       |
| 37  | ⼉       | 293        | 59.4%       |
| 38  | ⼥       | 288        | 60.2%       |
| 39  | ⾙       | 287        | 61.0%       |
| 40  | ⽬       | 282        | 61.7%       |
| 41  | ⼛       | 279        | 62.5%       |
| 42  | ⾡       | 277        | 63.2%       |
| 43  | ⽯       | 276        | 63.9%       |
| 44  | ⼡       | 264        | 64.6%       |
| 45  | ⼚       | 257        | 65.3%       |
| 46  | ⾐       | 249        | 66.0%       |
| 47  | ⽕       | 248        | 66.6%       |
| 48  | ⽲       | 248        | 67.3%       |
| 49  | ⼹       | 245        | 67.9%       |
| 50  | ⾞       | 233        | 68.5%       |
| 51  | ⾜       | 224        | 69.1%       |
| 52  | ⾺       | 219        | 69.7%       |
| 53  | ⾫       | 214        | 70.3%       |
| 54  | ⽟       | 209        | 70.8%       |
| 55  | ⽰       | 209        | 71.4%       |
| 56  | ⼽       | 208        | 71.9%       |
| 57  | ⼘       | 207        | 72.5%       |
| 58  | ⽁       | 201        | 73.0%       |
| 59  | ⼫       | 199        | 73.5%       |
| 60  | ⽶       | 195        | 74.1%       |
| 61  | ⾨       | 183        | 74.5%       |
| 62  | ⼭       | 179        | 75.0%       |
| 63  | ⽹       | 177        | 75.5%       |
| 64  | ⼐       | 174        | 75.9%       |
| 65  | ⼅       | 168        | 76.4%       |
| 66  | ⼨       | 166        | 76.8%       |
| 67  | ⼲       | 163        | 77.3%       |
| 68  | ⼒       | 162        | 77.7%       |
| 69  | ⾴       | 160        | 78.1%       |
| 70  | ⾌       | 157        | 78.5%       |
| 71  | ⼏       | 154        | 78.9%       |
| 72  | ⽻       | 153        | 79.3%       |
| 73  | ⼱       | 153        | 79.7%       |
| 74  | ⼠       | 153        | 80.1%       |
| 75  | ⼙       | 152        | 80.5%       |
| 76  | ⾷       | 151        | 80.9%       |
| 77  | ⼔       | 141        | 81.3%       |
| 78  | ⾬       | 141        | 81.7%       |
| 79  | ⼦       | 141        | 82.1%       |
| 80  | ⼎       | 137        | 82.4%       |
| 81  | ⽿       | 137        | 82.8%       |
| 82  | ⽴       | 137        | 83.2%       |
| 83  | ⼕       | 133        | 83.5%       |
| 84  | ⽝       | 132        | 83.9%       |
| 85  | ⽺       | 132        | 84.2%       |
| 86  | ⼯       | 126        | 84.5%       |
| 87  | ⽩       | 124        | 84.9%       |
| 88  | ⽖       | 124        | 85.2%       |
| 89  | ⾣       | 121        | 85.5%       |
| 90  | ⽌       | 119        | 85.8%       |
| 91  | ⼞       | 117        | 86.1%       |
| 92  | ⾅       | 116        | 86.4%       |
| 93  | ⽫       | 113        | 86.7%       |
| 94  | ⼴       | 113        | 87.0%       |
| 95  | ⽳       | 108        | 87.3%       |
| 96  | ⽧       | 100        | 87.6%       |
| 97  | ⾰       | 100        | 87.9%       |
| 98  | ⽋       | 99         | 88.1%       |
| 99  | ⽤       | 98         | 88.4%       |
| 100 | ⼺       | 98         | 88.6%       |
| 101 | ⾑       | 96         | 88.9%       |
| 102 | ⽜       | 93         | 89.1%       |
| 103 | ⾀       | 93         | 89.4%       |
| 104 | ⼸       | 91         | 89.6%       |
| 105 | ⽄       | 89         | 89.9%       |
| 106 | ⼰       | 87         | 90.1%       |
| 107 | ⽢       | 86         | 90.3%       |
| 108 | ⾈       | 85         | 90.5%       |
| 109 | ⽅       | 82         | 90.8%       |
| 110 | ⽎       | 80         | 91.0%       |
| 111 | ⼣       | 78         | 91.2%       |
| 112 | ⽮       | 78         | 91.4%       |
| 113 | ⽼       | 77         | 91.6%       |
| 114 | ⾖       | 72         | 91.8%       |
| 115 | ⼪       | 68         | 92.0%       |
| 116 | ⽭       | 68         | 92.1%       |
| 117 | ⼻       | 67         | 92.3%       |
| 118 | ⾤       | 65         | 92.5%       |
| 119 | ⾒       | 64         | 92.7%       |
| 120 | ⾽       | 64         | 92.8%       |
| 121 | ⼬       | 63         | 93.0%       |
| 122 | ⾥       | 61         | 93.2%       |
| 123 | ⽐       | 60         | 93.3%       |
| 124 | ⾟       | 60         | 93.5%       |
| 125 | ⽍       | 60         | 93.6%       |
| 126 | ⾗       | 60         | 93.8%       |
| 127 | ⽽       | 58         | 93.9%       |
| 128 | ⾮       | 56         | 94.1%       |
| 129 | ⽂       | 55         | 94.2%       |
| 130 | ⼖       | 55         | 94.4%       |
| 131 | ⾭       | 51         | 94.5%       |
| 132 | ⼶       | 50         | 94.6%       |
| 133 | ⿅       | 49         | 94.8%       |
| 134 | ⾛       | 49         | 94.9%       |
| 135 | ⼳       | 49         | 95.0%       |
| 136 | ⾓       | 49         | 95.2%       |
| 137 | ⽸       | 48         | 95.3%       |
| 138 | ⾕       | 47         | 95.4%       |
| 139 | ⽒       | 47         | 95.5%       |
| 140 | ⽈       | 47         | 95.7%       |
| 141 | ⽚       | 46         | 95.8%       |
| 142 | ⽱       | 45         | 95.9%       |
| 143 | ⼮       | 44         | 96.0%       |
| 144 | ⾉       | 44         | 96.1%       |
| 145 | ⼵       | 43         | 96.3%       |
| 146 | ⾘       | 42         | 96.4%       |
| 147 | ⾻       | 42         | 96.5%       |
| 148 | ⿒       | 40         | 96.6%       |
| 149 | ⾳       | 40         | 96.7%       |
| 150 | ⾏       | 38         | 96.8%       |
| 151 | ⾱       | 38         | 96.9%       |
| 152 | ⽏       | 37         | 97.0%       |
| 153 | ⿓       | 37         | 97.1%       |
| 154 | ⿁       | 35         | 97.2%       |
| 155 | ⾃       | 35         | 97.3%       |
| 156 | ⿈       | 33         | 97.4%       |
| 157 | ⿇       | 33         | 97.4%       |
| 158 | ⾚       | 33         | 97.5%       |
| 159 | ⽾       | 32         | 97.6%       |
| 160 | ⿊       | 32         | 97.7%       |
| 161 | ⽦       | 31         | 97.8%       |
| 162 | ⼷       | 31         | 97.9%       |
| 163 | ⾄       | 30         | 97.9%       |
| 164 | ⽪       | 30         | 98.0%       |
| 165 | ⾆       | 30         | 98.1%       |
| 166 | ⽑       | 29         | 98.2%       |
| 167 | ⾼       | 28         | 98.3%       |
| 168 | ⾠       | 28         | 98.3%       |
| 169 | ⽣       | 27         | 98.4%       |
| 170 | ⾵       | 27         | 98.5%       |
| 171 | ⾝       | 24         | 98.5%       |
| 172 | ⿏       | 24         | 98.6%       |
| 173 | ⽙       | 24         | 98.7%       |
| 174 | ⽓       | 24         | 98.7%       |
| 175 | ⽀       | 23         | 98.8%       |
| 176 | ⿀       | 23         | 98.8%       |
| 177 | ⿑       | 23         | 98.9%       |
| 178 | ⽨       | 22         | 99.0%       |
| 179 | ⽞       | 22         | 99.0%       |
| 180 | ⾇       | 21         | 99.1%       |
| 181 | ⾂       | 21         | 99.1%       |
| 182 | ⾧       | 21         | 99.2%       |
| 183 | ⾲       | 20         | 99.2%       |
| 184 | ⿎       | 18         | 99.3%       |
| 185 | ⽘       | 18         | 99.3%       |
| 186 | ⽠       | 17         | 99.4%       |
| 187 | ⿐       | 16         | 99.4%       |
| 188 | ⽆       | 16         | 99.5%       |
| 189 | ⽃       | 15         | 99.5%       |
| 190 | ⽡       | 15         | 99.6%       |
| 191 | ⿌       | 15         | 99.6%       |
| 192 | ⿕       | 14         | 99.6%       |
| 193 | ⽛       | 14         | 99.7%       |
| 194 | ⿆       | 14         | 99.7%       |
| 195 | ⾁       | 14         | 99.7%       |
| 196 | ⾎       | 12         | 99.8%       |
| 197 | ⽗       | 11         | 99.8%       |
| 198 | ⾊       | 10         | 99.8%       |
| 199 | ⾸       | 10         | 99.9%       |
| 200 | ⾹       | 9          | 99.9%       |
| 201 | ⾪       | 8          | 99.9%       |
| 202 | ⾯       | 7          | 99.9%       |
| 203 | ⾢       | 6          | 99.9%       |
| 204 | ⿍       | 5          | 99.9%       |
| 205 | ⼊       | 5          | 100.0%      |
| 206 | ⿄       | 4          | 100.0%      |
| 207 | ⿋       | 4          | 100.0%      |
| 208 | ⿔       | 3          | 100.0%      |
| 209 | ⾿       | 2          | 100.0%      |
| 210 | ⾶       | 1          | 100.0%      |
| 211 | ⿉       | 1          | 100.0%      |
| 212 | ⼾       | 1          | 100.0%      |
|     |         | 37789      |

